net/udprelay/metrics.go

```diff
@@ -0,0 +1,134 @@
+	cMetricForwarded66Packets = clientmetric.NewCounter("udprelay_forwarded_packets_udp6_udp6")
+
+	cMetricForwarded44Bytes = clientmetric.NewCounter("udprelay_forwarded_bytes_udp4_udp4")
+	cMetricForwarded46Bytes = clientmetric.NewCounter("udprelay_forwarded_bytes_udp4_udp6")
+	cMetricForwarded64Bytes = clientmetric.NewCounter("udprelay_forwarded_bytes_udp6_udp4")
+	cMetricForwarded66Bytes = clientmetric.NewCounter("udprelay_forwarded_bytes_udp6_udp6")
+
+	cMetricEndpoints = clientmetric.NewGauge("udprelay_endpoints")
+)
+
+type transport string
+
+const (
+	transportUDP4 transport = "udp4"
+	transportUDP6 transport = "udp6"
+)
+
+type forwardedLabel struct {
+	transportIn  transport `prom:"transport_in"`
+	transportOut transport `prom:"transport_out"`
+}
+
+type endpointLabel struct {
+}
+
+type metrics struct {
+	forwarded44Packets expvar.Int
+	forwarded46Packets expvar.Int
+	forwarded64Packets expvar.Int
+	forwarded66Packets expvar.Int
+
+	forwarded44Bytes expvar.Int
+	forwarded46Bytes expvar.Int
+	forwarded64Bytes expvar.Int
+	forwarded66Bytes expvar.Int
+
+	endpoints expvar.Int
+}
+
+// registerMetrics publishes user metric counters for peer relay server.
+func registerMetrics(reg *usermetric.Registry) *metrics {
+	var (
+		forwardedPackets = usermetric.NewMultiLabelMapWithRegistry[forwardedLabel](
+			reg,
+			"tailscaled_relay_forwarded_packets_total",
+			"counter",
+			"Number of packets forwarded via Peer Relay",
+		)
+		forwardedBytes = usermetric.NewMultiLabelMapWithRegistry[forwardedLabel](
+			reg,
+			"tailscaled_relay_forwarded_bytes_total",
+			"counter",
+			"Number of bytes forwarded via Peer Relay",
+		)
+		endpoints = usermetric.NewMultiLabelMapWithRegistry[endpointLabel](
+			reg,
+			"tailscaled_relay_endpoints_total",
+			"gauge",
+			"Number of allocated Peer Relay endpoints",
+		)
+		forwarded44 = forwardedLabel{transportIn: transportUDP4, transportOut: transportUDP4}
+		forwarded46 = forwardedLabel{transportIn: transportUDP4, transportOut: transportUDP6}
+		forwarded64 = forwardedLabel{transportIn: transportUDP6, transportOut: transportUDP4}
+		forwarded66 = forwardedLabel{transportIn: transportUDP6, transportOut: transportUDP6}
+		m           = new(metrics)
+	)
+
+	// Publish user metrics.
+	forwardedPackets.Set(forwarded44, &m.forwarded44Packets)
+	forwardedPackets.Set(forwarded46, &m.forwarded46Packets)
+	forwardedPackets.Set(forwarded64, &m.forwarded64Packets)
+	forwardedPackets.Set(forwarded66, &m.forwarded66Packets)
+
+	forwardedBytes.Set(forwarded44, &m.forwarded44Bytes)
+	forwardedBytes.Set(forwarded46, &m.forwarded46Bytes)
+	forwardedBytes.Set(forwarded64, &m.forwarded64Bytes)
+	forwardedBytes.Set(forwarded66, &m.forwarded66Bytes)
+
+	endpoints.Set(endpointLabel{}, &m.endpoints)
+
+	return m
+}
+
+// addEndpoints updates the total endpoints gauge. Value can be negative.
+func (m *metrics) addEndpoints(value int) {
+	m.endpoints.Add(int64(value))
+	cMetricEndpoints.Add(int64(value))
```

Comment:
Since the `expvar.Add()` API uses `int64`, we can do the same here. The `metrics.addEndpoints` call sites don't need any casting either since they are constants.